Separated javascript

dcangulo opened the pull request:
Created different javascript file for each view. Moved the javascript codes from application.js to specific javascript files. @juliusb0205

juliusb0205 commented on app/assets/javascripts/editpost.js:
```diff
@@ -0,0 +1,14 @@
+$(document).ready(function(){
+  authenticity_token = $("#authenticity_token").val();
```
Inexplicitly declared variable.
Use Double Quotes only when needed -> use $('#authenticity_token')

juliusb0205 commented on app/assets/javascripts/editpost.js:
```diff
@@ -0,0 +1,14 @@
+$(document).ready(function(){
+  authenticity_token = $("#authenticity_token").val();
+  $("#editpostform").on("submit", function(event){
```
Single Qoutes

juliusb0205 commented on app/assets/javascripts/editpost.js:
```diff
@@ -0,0 +1,14 @@
+  authenticity_token = $("#authenticity_token").val();
+  $("#editpostform").on("submit", function(event){
+    $.ajax({
+      type: "post",
```
Single Qoutes

juliusb0205 commented on app/assets/javascripts/index.js:
```diff
@@ -0,0 +1,66 @@
+$(document).ready(function(){
+  authenticity_token = $("#authenticity_token").val();
```
Inexplicitly declared variable.
Use Double Quotes only when needed -> use $('#authenticity_token')

juliusb0205 commented on app/assets/javascripts/index.js:
```diff
@@ -0,0 +1,66 @@
+  $(".pagination > a").addClass("page-item");
+  $(".pagination > a").addClass("page-link");
+  $("#postform").on("submit", function(event){
+    $.ajax({
```
Single Quotes above

juliusb0205 commented on app/assets/javascripts/index.js:
```diff
@@ -0,0 +1,66 @@
+        url: "/ajax/like",
+        data: {
+          authenticity_token:authenticity_token,
+          post_id:this.value
```
Space after :

juliusb0205 commented on app/assets/javascripts/index.js:
```diff
@@ -0,0 +1,66 @@
+    }
+    else {
+      $.ajax({
+        type: "post",
```
Single Quotes

juliusb0205 commented on app/views/posts/edit.haml:
```diff
@@ -1,3 +1,4 @@
+= javascript_include_tag "editpost"
```
Single quotes

juliusb0205 commented on app/views/posts/userlist.haml:
```diff
@@ -1,3 +1,4 @@
+= javascript_include_tag "users"
```
Single quotes

juliusb0205 commented on app/views/posts/index.haml:
```diff
@@ -1,3 +1,4 @@
+= javascript_include_tag "index"
```
Single Quotes

juliusb0205 commented on app/assets/javascripts/users.js:
```diff
@@ -0,0 +1,35 @@
+      type: "post",
+      url: "/ajax/unfollow",
+      data: {
+        authenticity_token:authenticity_token,
```
Space after :

juliusb0205 commented on app/views/posts/userlist.haml:
```diff
@@ -1,3 +1,4 @@
+= javascript_include_tag "users"
 %h2 People you may know
 %br/
 - @users.all.each do |user|
```
Check if .all can be omitted. If i'm not mistaken @users is already a collection in your controller which could be iterated by .each

dcangulo commented:
@kencruz07

kencruz07 commented on app/assets/javascripts/application.js:
```diff
@@ -2,156 +2,116 @@
-  $(".btn-delete-post").on("click", function(event) {
+  var authenticity_token = $('#authenticity_token').val();
+  $('.btn-delete-post').on('click', function(event) {
     var id = this.value;
```
Inside code blocks, insert a line break/new line after variable declarations
```javascript
var id = this.value;

$.ajax ...
```

kencruz07 commented on app/assets/javascripts/application.js:
```diff
@@ -2,156 +2,116 @@
-    var content = $("#commenton"+this.value).val();
+  $('.btn-submit-comment').on('click', function(event) {
+    var content = $('#commenton'+this.value).val();
     var id = this.value;
```
Inside code blocks, insert a line break/new line after variable declarations
```javascript
var id = this.value;

$.ajax ...
```

kencruz07 commented on app/assets/javascripts/application.js:
```diff
@@ -2,156 +2,116 @@
-    event.preventDefault();
-  });
-  $(".btn-delete-post").on("click", function(event) {
+  var authenticity_token = $('#authenticity_token').val();
```
Use `camelCase` naming for javascript variables

kencruz07 commented on app/assets/javascripts/application.js:
```diff
@@ -2,156 +2,116 @@
       },
       success: function() {
-       $("#post"+id).addClass("hidden");
+       $('#post'+id).addClass('hidden');
```
When using operators, separate operands with spaces
```javascript
$('#post' + id)
```

kencruz07 commented on app/assets/javascripts/application.js:
```diff
@@ -2,156 +2,116 @@
-        post_id:this.value,
+        authenticity_token: authenticity_token,
+        post_id: this.value,
         content:content
```
Separate assignments and values with spaces
```javascript
content: content
```

kencruz07 commented on app/assets/javascripts/application.js:
```diff
@@ -2,156 +2,116 @@
+  $('.like-button').on('click', function(event) {
+    var likebtn = $(this);
+    var id = this.value;
+    if(likebtn.hasClass('like-button-active')) {
```
for `if`, `for` statements, follow the spacing and convention. Please do mind the spaces:
```javascript
if ( condition && (condition || condition) ) {
...
}
else {
...
}
```

kencruz07 commented on app/controllers/application_controller.rb:
```diff
@@ -1,6 +1,12 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
```
While some parts of the Rails default code are auto generated, convert `key: :value` notation to our hash rocket convention: 
```ruby
:key => :value
```
Do this especially when editing a file with said notation

kencruz07 commented on app/controllers/posts_controller.rb:
```diff
@@ -30,6 +31,7 @@ def postprocess
 
   def show
     @posts = Post.where :id => params[:id]
+    @postdecorator = PostDecorator.new(@posts)
```
You can omit the parenthesis here. Loose rule when doing this is when this is the only operation done UPON assigning to a variable. You can do multiple parameters without parenthesis, but not cases like this: 
```ruby
@posts = PostDecorator.new @posts.where '...'
```
Above code will raise an error since multiple operations are inside. To fix:
```ruby
@posts = PostDecorator.new @posts.where('...')
```
*Notes: Just omit the top level possible parenthesis. Don't omit if the line exceeds max characters per line.

kencruz07 commented on app/controllers/users_controller.rb:
```diff
@@ -13,13 +14,25 @@ def registerprocess
-    @posts = Post.where(:user_id => @user.id).order 'created_at DESC'
+    user = User.find_by_id params[:id]
+    @user = UserDecorator.new user
+    @posts = Post.where(:user_id => params[:id]).order 'created_at DESC'
```
Add line break before returning a value
```ruby
@posts = ...

usercount
```

kencruz07 commented on app/controllers/users_controller.rb:
```diff
@@ -13,13 +14,25 @@ def registerprocess
     params.require(:user).permit :name, :username, :email, :password
   end
+
+  def layouts
```
If you are expecting only 2 values for this, it's better to use this instead:
```ruby
action_name == 'show' ? 'post' : 'application'
```

kencruz07 commented on app/decorators/post_decorator.rb:
```diff
@@ -0,0 +1,37 @@
+  delegate_all
+
+  def time_ago time
+    h.time_ago_in_words(time) + " ago"
```
When combining variable values and strings, use:
```ruby
"#{h.time_ago_in_words(time)} ago"
```

kencruz07 commented:
@dcangulo Please check comments. You may ask questions for clarifications about said rules/conventions. Also, some comments by @juliusb0205 are still unresolved. Please double check PR. 

Thanks. 👍

dcangulo commented:
@kencruz07 @juliusb0205 
*Fixed Double quotes to single quotes
*Added a new line after variable declarations
*Used camelCase for javascript variables
*Added a new line before returning values
*Added spaces when using operators
*Added spaces when using if .. else .. statements
*Used the shorthand for layouts
*Changed key: :value to :key => :value
*Removed parenthesis on the last parameter

juliusb0205 commented on app/assets/javascripts/application.js:
```diff
@@ -2,156 +2,122 @@
-    $("#comments"+this.value).addClass("hidden");
+  $('.like-button').on('click', function(event) {
+    var likebtn = $(this);
+    var id = this.value;
```
var likebtn = $(this),
      id = this.value;


For successive variable declaration follow the , syntax.

juliusb0205 commented on app/controllers/application_controller.rb:
```diff
@@ -1,5 +1,11 @@
+    @postcount = Post.where(:user_id => session[:user_id]).count
+    @followercount = Follow.where(:followed_id => session[:user_id]).count
+    @followingcount = Follow.where(:follower_id => session[:user_id]).count
+  end
```
Since current user is defined try

```
@postcount = Post.where(:user => current_user).count

```

juliusb0205 commented on app/controllers/likes_controller.rb:
```diff
@@ -2,15 +2,17 @@ class LikesController < ApplicationController
-      likecount: Like.where(:post_id => params[:post_id]).count
+      :likecount => Like.where(:post_id => params[:post_id]).count
     }
   end
```
Update this file to use current_user

juliusb0205 commented on app/controllers/posts_controller.rb:
```diff
@@ -9,32 +9,38 @@ def index
   end
 
   def postprocess
     post = Post.new :content => params[:post_content], :user_id => session[:user_id]
```
Use current_user

juliusb0205 replied:
Use Post.create if there are no more transformation to data.

.create is a combination of .new and .save at the same time.

juliusb0205 commented on app/controllers/posts_controller.rb:
```diff
@@ -9,32 +9,38 @@ def index
   def postprocess
     post = Post.new :content => params[:post_content], :user_id => session[:user_id]
     post.save
     post_id = nil
```
Unused assignment. Confirm it this can be removed.

juliusb0205 commented on app/controllers/posts_controller.rb:
```diff
@@ -9,32 +9,38 @@ def index
     @latest.each do |lpost|
       post_id = lpost.id
     end
+
```
@latest selects the most recent post.
Most recent post is already defined on line 19 after running post.save
Remove this block. If not needed.

Use this below
 ```
:postid => post.id
```

juliusb0205 commented on app/controllers/posts_controller.rb:
```diff
@@ -51,44 +57,39 @@ def postcomment
     @latest.each do |lcomment|
       comment_id = lcomment.id
     end
+
```
Same comment above.
If this block is used to get the latest comment, use the declared comment.
Remove if not needed.

juliusb0205 commented on app/controllers/posts_controller.rb:
```diff
@@ -51,44 +57,39 @@ def postcomment
-      commentid: comment_id,
-      content: params[:content]
+      :username => session[:username],
+      :commentid => comment_id,
```
Use comment.id

juliusb0205 commented on app/controllers/posts_controller.rb:
```diff
@@ -51,44 +57,39 @@ def postcomment
+
     usercount
   end
 end
```
Remove usercount if not being used.

juliusb0205 commented on app/decorators/follow_decorator.rb:
```diff
@@ -0,0 +1,24 @@
+    end
+  end
+
+end
```
Use ternary operator
If this cant fit line store condition in a variable

```
condition ? value_if_true : value_if_false
```

juliusb0205 commented on app/views/posts/userlist.haml:
```diff
@@ -5,10 +5,7 @@
-            %strong
-              = user.name
-            @#{user.username}
+          %strong= link_to "#{user.name}</strong> (@#{user.username})".html_safe, user_path(user)
```
Use link_to do block

juliusb0205 commented on app/views/posts/viewfollowers.haml:
```diff
@@ -1,14 +1,13 @@
+            = user.name
+          @#{user.username}
+        .col-md-6.text-right
+          - if Follow.where(:follower_id => session[:user_id], :followed_id => user.id).present?
```
Use current_user.followers.present?

juliusb0205 commented on app/controllers/posts_controller.rb:
```diff
@@ -3,39 +3,33 @@ class PostsController < ApplicationController
+    @postdecorator = PostDecorator.new @posts
   end
 
   def postprocess
```
Convention method names in ruby that have more than one word should be all lowercase and joined by underscore.

```
def this_is_a_method_name
end
```

juliusb0205 commented on app/controllers/posts_controller.rb:
```diff
@@ -3,39 +3,33 @@ class PostsController < ApplicationController
-    usercount
   end
 
   def editpostprocess
```
Use method name convention

juliusb0205 commented on app/controllers/posts_controller.rb:
```diff
@@ -48,47 +42,28 @@ def deletepost
   end
 
   def postcomment
```
Use method name convention

juliusb0205 commented on app/controllers/posts_controller.rb:
```diff
@@ -48,47 +42,28 @@ def deletepost
     }
   end
 
   def deletecomment
```
Use method name convention

juliusb0205 commented on app/controllers/posts_controller.rb:
```diff
@@ -48,47 +42,28 @@ def deletepost
-    @followingcount = Follow.where(:follower_id => session[:user_id]).count
   end
 
   def viewfollowers
```
Use method name convention

juliusb0205 commented on app/controllers/posts_controller.rb:
```diff
@@ -48,47 +42,28 @@ def deletepost
+    @followers = current_user.followers
   end
 
   def viewfollowing
```
Use method name convention

kencruz07 commented on app/assets/javascripts/application.js:
```diff
@@ -2,156 +2,122 @@
-  $(".pagination > a").addClass("page-link");
-  $(".like-button").on("click", function(event) {
-    var likebtn = $(this);
+  var authenticityToken = $('#authenticity_token').val();
```
Use dashes instead of underscores for HTML id and class naming

kencruz07 commented on app/controllers/likes_controller.rb:
```diff
@@ -1,16 +1,19 @@
+
     render json: {
-      likecount: Like.where(:post_id => params[:post_id]).count
+      :likecount => post.users.count
```
Separate words with underscores, `:like_count`

kencruz07 commented on app/controllers/posts_controller.rb:
```diff
@@ -48,47 +42,33 @@ def deletepost
-      commentid: comment_id,
-      content: params[:content]
+      :username => session[:username],
+      :commentid => comment.id,
```
Separate words with underscore

kencruz07 commented on app/views/layouts/post.haml:
```diff
@@ -1,25 +1,27 @@
     = csp_meta_tag
-    = stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track': 'reload'
+    = stylesheet_link_tag    'application',
+      media: 'all', 'data-turbolinks-track': 'reload'
```
Follow hash rocket notation

kencruz07 commented on app/views/layouts/post.haml:
```diff
@@ -1,25 +1,27 @@
   %body
     %nav.navbar.navbar-expand-md.fixed-top
-      %button.navbar-toggler{"data-target" => "#collapsibleNavbar", "data-toggle" => "collapse", :type => "button"}
+      %button.navbar-toggler{'data-target' => '#collapsibleNavbar',
```
Use dash separators to separate HTML ids and classes

kencruz07 commented on app/controllers/posts_controller.rb:
```diff
@@ -3,39 +3,33 @@ class PostsController < ApplicationController
-    followed.push session[:user_id]
-    @posts = Post.where(:user_id => followed).order("created_at DESC").paginate(:page => params[:page], :per_page => 10)
-    usercount
+    followings = current_user.followings.map &:id
```
Try to rename all instances of "Followings" to just "Following"

juliusb0205 commented on app/views/posts/view_followers.haml:
```diff
@@ -0,0 +1,15 @@
+            = user.name
+          @#{user.username}
+        .col-md-6.text-right
+          - if current_user.following.map(&:id).include? user.id
```
create a method in your model named 'followed_by'.

Method should return true or false depending on the condition if the user is being followed by the user passed in the argument.

```
def followed_by user

end
```

Use this new method on the view.

juliusb0205 commented on app/views/posts/user_list.haml:
```diff
@@ -0,0 +1,17 @@
+            %strong= user.name
+            (@#{user.username})
+        .col-md-6.text-right
+          - if current_user.following.map(&:id).include? user.id
```
Check 'followed_by' comment/request below.

juliusb0205 commented on app/views/sessions/login.haml:
```diff
@@ -1,6 +1,6 @@
+  %input{:type => 'password', :name => 'password'}
   = hidden_field_tag :authenticity_token, form_authenticity_token
-  %button{:type => "submit"} Submit
+  %button{:type => 'submit'} Submit
```
Use form_for

Look at your register.haml file, just copy and update it accordingly.
Major change should be the url it is posting to.
